tests/kernel/smp/src/main.c

```diff
@@ -14,11 +14,12 @@
 #error SMP test requires at least two CPUs!
 #endif
 
+#define MAX_NUM_THREADS CONFIG_MP_MAX_NUM_CPUS
 #define RUN_FACTOR (CONFIG_SMP_TEST_RUN_FACTOR / 100.0)
 
 #define T2_STACK_SIZE (2048 + CONFIG_TEST_EXTRA_STACK_SIZE)
 #define STACK_SIZE (384 + CONFIG_TEST_EXTRA_STACK_SIZE)
-#define DELAY_US 50000
+#define DELAY_US (50000 * MAX_NUM_THREADS)
```

Comment:
This seems suspect.  50ms is a **very long time** already.  Surely it doesn't take 50ms per core to do anything in this test.  Would prefer to see whatever this failure is root caused.  This looks like a hack to me.

Reply:
Removed the changes